controllers/flinkcluster/flinkcluster_observer.go

```diff
@@ -277,10 +277,10 @@ func (observer *ClusterStateObserver) observeJob(
 	}
 
 	var submitterLog *SubmitterLog
-	// Extract submission result only when it is in deployment progress.
+	// Extract submission result only when it is in deployment progress or the submitter pod failed.
 	// It is not necessary to get the log stream from the submitter pod always.
 	var jobDeployInProgress = recordedJob != nil && recordedJob.State == v1beta1.JobStateDeploying
-	if jobPod != nil && jobDeployInProgress {
+	if jobPod != nil && (jobDeployInProgress || jobPod.Status.Phase == corev1.PodFailed) {
```

Comment:
Could we achieve the same thing if we rely on the recordedJob.State == v1beta1.JobStateFailed? from a first looks it looks like it.

Reply:
This check happens before recordedJob.State is updated. That is why I check pod status directly

Reply:
It's ok to check pod, but to be clear, observer should run again once the state is updated to JobStateFailed; That's why I think it would work as well.

Reply:
I see. Up to you then. If you want me to change it, please let me know. I can run another test and if all good update the PR.